hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountInfoHandler.java

```diff
@@ -58,13 +106,289 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final CryptoGetInfoQuery op = query.cryptoGetInfoOrThrow();
+        if (op.hasAccountID()) {
+            final var account = accountStore.getAccountById(requireNonNull(op.accountID()));
+            validateFalsePreCheck(account == null, INVALID_ACCOUNT_ID);
+            validateFalsePreCheck(account.deleted(), ACCOUNT_DELETED);
+        } else {
+            throw new PreCheckException(INVALID_ACCOUNT_ID);
+        }
     }
 
     @Override
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final var tokenRelationStore = context.createStore(ReadableTokenRelationStore.class);
+        final var tokenStore = context.createStore(ReadableTokenStore.class);
+        final var stakingInfoStore = context.createStore(ReadableStakingInfoStore.class);
+        final var op = query.cryptoGetInfoOrThrow();
+        final var response = CryptoGetInfoResponse.newBuilder();
+        final var accountId = op.accountIDOrElse(AccountID.DEFAULT);
+
+        response.header(header);
+        final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
+        if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
+            final var optionalInfo = infoForAccount(
+                    accountId,
+                    accountStore,
+                    tokenStore,
+                    tokenRelationStore,
+                    stakingInfoStore,
+                    tokensConfig,
+                    ledgerConfig,
+                    new RewardCalculator() {
+                        @Override
+                        public long epochSecondAtStartOfPeriod(long stakePeriod) {
+                            return 0;
+                        }
+
+                        @Override
+                        public long estimatePendingRewards(Account account, @Nullable StakingNodeInfo stakingNodeInfo) {
+                            return 0;
+                        }
+                    }); // remove this when we have a real reward calculator
+            if (optionalInfo.isPresent()) {
+                response.accountInfo(optionalInfo.get());
```

Comment:
Don't you need to set `.responseType(ANSWER_ONLY)` here ?

Reply:
Will make the change in TokenGetInfoHandler pr.